pkg/downloader/chart_downloader.go

```diff
@@ -94,12 +95,31 @@ func (c *ChartDownloader) DownloadTo(ref, version, dest string) (string, *proven
 		return "", nil, err
 	}
 
-	data, err := g.Get(u.String(), c.Options...)
+	downloadURL := u.String()
+	name := filepath.Base(u.Path)
+
+	if _, ok := g.(*registry.Getter); ok {
+		parts := strings.Split(filepath.Base(u.Path), ":")
+
+		if len(parts) == 1 && version == "" {
+			return "", nil, errors.New("no version or tag provided")
+		}
+
+		if len(parts) != 2 {
+			parts = append(parts, version)
+			u.Path = fmt.Sprintf("%s:%s", u.Path, version)
+		}
+
+		downloadURL = u.String()
+		name = fmt.Sprintf("%s-%s.tgz", parts[0], parts[1])
```

Comment:
Giving the same example:

```
repository: oci://localhost:5000/testrepo/testchart:1.0.1:1.2.3
version: 1.0.0
```

How is this considered valid input? How is the chart downloader able to determine that `parts[1]` is the right version?

This makes me go back to the original question: why are we accepting tags in the URL path in the first place?

Reply:
Yep that shouldn't be valid, as that `repository` tag is invalid. As mentioned in the other comment – I can add validation for that if desired.

As to why we're accepting them: supporting the same format as `helm chart save` (as you mentioned in https://github.com/helm/helm/issues/8332#issuecomment-648305142). I think it's perfectly fine to drop support for arbitrary tags, but then we should do that for `helm chart` operations as well.